ambassador/ambassador/envoy/v2/v2listener.py

```diff
@@ -27,34 +27,41 @@
 if TYPE_CHECKING:
     from . import V2Config
 
-
-ExtAuthRequestHeaders = {
-    'Authorization': True,
-    'Cookie': True,
-    'Forwarded': True,
-    'From': True,
-    'Host': True,
-    'Proxy-Authenticate': True,
-    'Proxy-Authorization': True,
-    'Set-Cookie': True,
-    'User-Agent': True,
-    'X-Forwarded-For': True,
-    'X-Forwarded-Host': True,
-    'X-Forwarded-Proto'
-    'X-Gateway-Proto': True,
-    'WWW-Authenticate': True,
-}
+# Static header keys normally used in the context of and authorization request.
+AllowedRequestHeaders = frozenset([
+    'authorization',
+    'cookie',
+    'from',
+    'host',
+    'proxy-authorization',
+    'user-agent',
+    'x-forwarded-for',
+    'x-forwarded-host',
+    'x-forwarded-proto'
+])
+
+# Static header keys normally used in the context of and authorization response.
+AllowedAuthorizationHeaders = frozenset([
+    'location',
+    'authorization',
+    'proxy-authenticate',
+    'set-cookie',
+    'www-authenticate'
+])
 
 @multi
 def v2filter(irfilter):
     return irfilter.kind
 
 @v2filter.when("IRAuth")
 def v2filter(auth):
-    request_headers = dict(ExtAuthRequestHeaders)
-
-    for hdr in auth.allowed_headers:
-        request_headers[hdr] = True
+    if auth.api_version == "ambassador/v1":
+        authorizarion_headers = set(auth.allowed_authorization_headers)
+        request_headers = set(auth.allowed_request_headers)
+                
+    if auth.api_version == "ambassador/v0":
+        authorizarion_headers = set(auth.allowed_headers)
+        request_headers = authorizarion_headers
 
```

Comment:
We should probably post an error here if the `api_version` is neither `v0` nor `v1`. (Mind you, I'm not actually sure this is the right place for that -- maybe better done in the IR?

(Also guessing `authorizarion` is unintentional...)